drivers/mbox/mbox_nrf_vevif_event_local.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrf_vevif_event_local
+
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+
+#include <hal/nrf_vpr.h>
+
+/* callbacks */
+struct mbox_vevif_event_local_cbs {
+	mbox_callback_t cb[NRF_VPR_EVENTS_TRIGGERED_COUNT];
+	void *user_data[NRF_VPR_EVENTS_TRIGGERED_COUNT];
+	uint32_t enabled_mask;
+};
+
+struct mbox_vevif_event_local_conf {
+	NRF_VPR_Type *vpr;
+	uint32_t events_mask;
+	uint8_t events;
+	void (*irq_connect)(void);
+};
+
+static void vevif_event_local_isr(const void *device)
+{
+	const struct device *dev = (struct device *)device;
+	const struct mbox_vevif_event_local_conf *config = dev->config;
+	struct mbox_vevif_event_local_cbs *cbs = dev->data;
+
+	for (uint8_t id = 0; id < NRF_VPR_EVENTS_TRIGGERED_COUNT; id++) {
+		nrf_vpr_event_t event = nrf_vpr_triggered_event_get(id);
+		if ((cbs->enabled_mask & BIT(id)) && nrf_vpr_event_check(config->vpr, event)) {
```

Comment:
I'm curious if this condition can lead to some kind of a IRQ loop.

Let's assume there is an interrupt enabled for event X by `nrf_vpr_int_enable(config->vpr, BIT(X));`, but bit X is cleared in `cbs->enabled_mask`.
1. Event X is triggered
2. Event X causes IRQ in ARM
3. ARM handles IRQ with `vevif_event_local_isr()`
4. ARM iterates over all events, but this condition evaluates to `false` for all iterations
5. ARM exits the loop and the IRQ handler
6. The event is still pending, so it causes IRQ in ARM (go to step 2)

Reply:
On the other hand, the scenario in which my assumption is present is invalid. So maybe we can let the watchdog handle the problem